packages/core/src/test/amazonq/customizationUtil.test.ts

```diff
@@ -0,0 +1,99 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as sinon from 'sinon'
+import assert from 'assert'
+import { tryRegister } from '../testUtil'
+import {
+    amazonQScopes,
+    AuthUtil,
+    baseCustomization,
+    Customization,
+    FeatureConfigProvider,
+    getSelectedCustomization,
+    refreshStatusBar,
+    setSelectedCustomization,
+} from '../../codewhisperer'
+import { FeatureContext, globals } from '../../shared'
+import { resetCodeWhispererGlobalVariables } from '../codewhisperer/testUtil'
+import { createSsoProfile, createTestAuth } from '../credentials/testUtil'
+import { SsoConnection } from '../../auth'
+
+const enterpriseSsoStartUrl = 'https://enterprise.awsapps.com/start'
+
+describe('CodeWhisperer-customizationUtils', function () {
+    let auth: ReturnType<typeof createTestAuth>
+    let ssoConn: SsoConnection
+
+    before(async function () {
+        createTestAuth(globals.globalState)
+        tryRegister(refreshStatusBar)
+    })
+
+    beforeEach(async function () {
+        auth = createTestAuth(globals.globalState)
+        ssoConn = await auth.createInvalidSsoConnection(
+            createSsoProfile({ startUrl: enterpriseSsoStartUrl, scopes: amazonQScopes })
+        )
+        await resetCodeWhispererGlobalVariables()
+    })
+
+    afterEach(function () {
+        sinon.restore()
+    })
+
+    it('Returns baseCustomization when not SSO', async function () {
+        sinon.stub(AuthUtil.instance, 'isConnectionExpired').returns(false)
+        sinon.stub(AuthUtil.instance, 'isConnected').returns(true)
+        sinon.stub(AuthUtil.instance, 'isValidEnterpriseSsoInUse').returns(false)
+        sinon.stub(AuthUtil.instance, 'isCustomizationFeatureEnabled').value(true)
+        const customization = getSelectedCustomization()
+
+        assert.strictEqual(customization.name, baseCustomization.name)
+    })
+
+    it('Returns selectedCustomization when customization manually selected', async function () {
+        sinon.stub(AuthUtil.instance, 'isConnectionExpired').returns(false)
+        sinon.stub(AuthUtil.instance, 'isConnected').returns(true)
+        sinon.stub(AuthUtil.instance, 'isValidEnterpriseSsoInUse').returns(true)
+        sinon.stub(AuthUtil.instance, 'isCustomizationFeatureEnabled').value(true)
+        sinon.stub(AuthUtil.instance, 'conn').value(ssoConn)
+
+        const selectedCustomization: Customization = {
+            arn: 'testCustomizationArn',
+            name: 'testCustomizationName',
+            description: 'testCustomizationDescription',
+        }
+
```

Comment:
i feel it's better to have `sinon.stub(FeatureConfigProvider, 'getFeature').returns(featureCustomization)` here as well

Reply:
to simulate the real case when there is an override and user pre-selected customization

Reply:
Added to beforeEach so it's present for all tests